pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5520,3 +5520,164 @@ func TestRhdmProdImmutableEnvironmentWithoutJbpmCluster(t *testing.T) {
 	assert.Equal(t, int32(1), env.Servers[0].DeploymentConfigs[0].Spec.Replicas)
 	assert.Nil(t, cr.Status.Applied.Objects.Console, "Console should be nil")
 }
+
+func TestRhdmEnvironmentWithKafkaExt(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						Kafka: createKafkaExtObject(),
+					},
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.True(t, len(cr.Spec.Objects.Servers[0].Kafka.Topics) == 2)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	autocreateEnv, maxBlockMS, topics, extEnabled, clientID, acks, groupID, bootstrapServers := false, false, false, false, false, false, false, false
+	for _, env := range envs {
+
+		switch e := env.Name; e {
+		case "KIE_SERVER_KAFKA_EXT_ENABLED":
+			extEnabled = true
+			assert.True(t, env.Value == "true")
+		case "KIE_SERVER_KAFKA_EXT_GROUP_ID":
+			groupID = true
+			assert.True(t, env.Value == "my-kafka-group")
+
+		case "KIE_SERVER_KAFKA_EXT_ACKS":
+			acks = true
+			assert.True(t, env.Value == "2")
+
+		case "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS":
+			autocreateEnv = true
+			assert.True(t, env.Value == "true")
+
+		case "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS":
+			maxBlockMS = true
+			assert.True(t, env.Value == "2100")
+
+		case "KIE_SERVER_KAFKA_EXT_CLIENT_ID":
+			clientID = true
+			assert.True(t, env.Value == "C1234567")
+
+		case "KIE_SERVER_KAFKA_EXT_SERVERS":
+			bootstrapServers = true
+			assert.True(t, env.Value == "localhost:9092")
+
+		case "KIE_SERVER_KAFKA_EXT_TOPICS":
+			topics = true
+			assert.True(t, env.Value == "events=my-topics,errors=my-errs")
+		}
+	}
+	assert.True(t, autocreateEnv)
```

Comment:
not sure if I understood the purpose of these assertions here.

Reply:
Checks if all envs are present with the expected values

Reply:
see my comment about, there is a better way to do this.

Reply:
yep